velox/expression/SimpleFunctionAdapter.h

```diff
@@ -600,15 +600,23 @@ class SimpleFunctionAdapter : public VectorFunction {
           });
         }
       } else if (allNotNull) {
-        applyContext.applyToSelectedNoThrow([&](auto row) INLINE_LAMBDA {
-          // Passing a stack variable have shown to be boost the performance
-          // of functions that repeatedly update the output. The opposite
-          // optimization (eliminating the temp) is easier to do by the
-          // compiler (assuming the function call is inlined).
-          typename return_type_traits::NativeType out{};
-          bool notNull = doApplyNotNull<0>(row, out, readers...);
-          writeResult(row, notNull, out);
-        });
+        if (applyContext.allAscii) {
```

Comment:
@PHILO-HE  internal meta tools reported build time and size regressions can you wrap this part in a constexpr
like this:
this way the compilation of the added loop will be triggrered only when callAscii is defined, would be nice to do the same for the other usage. t
```
 else if (allNotNull) {
        if constexpr (FUNC::has_ascii) {
          if (applyContext.allAscii) {
            applyContext.applyToSelectedNoThrow([&](auto row) INLINE_LAMBDA {
              typename return_type_traits::NativeType out{};
              bool notNull = doApplyAsciiNotNull<0>(row, out, readers...);
              writeResult(row, notNull, out);
            });
            return;
          }
        }

        applyContext.applyToSelectedNoThrow([&](auto row) INLINE_LAMBDA {
          // Passing a stack variable have shown to be boost the performance
          // of functions that repeatedly update the output. The opposite
          // optimization (eliminating the temp) is easier to do by the
          // compiler (assuming the function call is inlined).
          typename return_type_traits::NativeType out{};
          bool notNull = doApplyNotNull<0>(row, out, readers...);
          writeResult(row, notNull, out);
        });
      } 
      ```

Reply:
Just added the check for `has_ascii` for the two usages. Thanks!